### PR TITLE
remove request and offer options

### PR DESCRIPTION
**Pull request by jerrett**

### CODE REVIEW

**Comment by rfb** [comment]
I would like to use an interaction that withdraws requests and offers rather then deleting. The main reason behind that trackability, like if someone put up a spam offer it would be nice if we could find it later - but of course, if the user can just remove the details using the edit form we lose that information.

Thoughts?

**Comment by jerrett** [comment]
it'd have to still appear like a delete since users have 1 offer/request , but we could mark it as "deleted"  instead of deleting it

**Comment by hez** [comment]
Ryan and I were talking about implementing a soft delete, this would probably solve that problem.

**Comment by jerrett** [comment]
maybe via https://github.com/rubysherpas/paranoia ?

**Comment by rfb** [comment]
One approach I'm considering is to add a `withdrawn` flag that could be set on an Offer or a Request, which would remove the Offer from listings and prevent Requests from receiving notifications (in the future..?)

This bring up the question of what we'll do when a user goes to add a new request or offer. I think basically the user would land on the edit page and it would ask if they wanted to publish their offer/request again or make a new offer/request (which would just drop them on a clear version of the form?)

**Comment by jerrett** [comment]
@rfb withdrawn and re-instate added

File: app/models/request.rb [hunk 1]
```diff
@@ -4,6 +4,10 @@ class Request < ApplicationRecord
   include Categorizable
   include PostalCodeable
 
+  belongs_to :user
+
+  enum status: %i[active withdrawn]
```
Comment (hez) [inline app/models/request.rb]: We should map these explicitly

` enum status: {active: 1, withdrawn: 2}`

File: app/views/offers/_form.html.erb [hunk 2]
```diff
@@ -22,7 +22,7 @@
 
 <div class="form-group <%= 'error' if @offer.errors.has_key?(:has_contact_information) %>">
   <%= form.label :email %>
-  <%= form.text_field :email, class: "form-control" %>
+  <%= form.text_field :email, class: "form-control", value: current_user.email %>
```
Comment (rfb) [inline app/views/offers/_form.html.erb]: again reading my mind

**Comment by jerrett** [comment]
made status mapping explict - plz merge

**Pull request merged**